nautobot_floor_plan/filter_extensions.py

```diff
@@ -1,9 +1,80 @@
 """Extensions to Nautobot core models' filtering functionality."""
 
+import logging
+
 import django_filters
 from nautobot.apps.filters import FilterExtension, RelatedMembershipBooleanFilter
 
 from nautobot_floor_plan import models
+from nautobot_floor_plan.utils.label_converters import LabelToPositionConverter, PositionToLabelConverter
+
+logger = logging.getLogger(__name__)
+
+
+class FloorPlanCoordinateFilter(django_filters.CharFilter):
+    """Custom filter that converts between human-readable labels and stored coordinates."""
+
+    def __init__(self, *, axis, **kwargs):
+        """Overwrite the constructor to set initial values."""
+        super().__init__(**kwargs)
+        self.axis = axis
+
+    def filter(self, qs, value):
+        """Convert label to position and filter."""
+        floor_plan_id = None
+        if not value:
+            return qs
+
+        try:
+            if hasattr(self, "parent") and self.parent and hasattr(self.parent, "data"):
+                floor_plan_id = self.parent.data.get("nautobot_floor_plan_floor_plan")
+            if not floor_plan_id:
+                return qs
+
+            floor_plan = models.FloorPlan.objects.get(pk=floor_plan_id)
+
+            # Check if custom labels are being used for this axis
+            if floor_plan.custom_labels.filter(axis=self.axis).exists():
+                # Use LabelToPositionConverter for custom labels
+                converter = LabelToPositionConverter(value, self.axis, floor_plan)
+                position, _ = converter.convert()
+            else:
+                # For default labels, use the value directly
+                position = value
+
+            return super().filter(qs, position)
+
+        except models.FloorPlan.DoesNotExist as e:
+            logger.error("Floor plan not found: %s", str(e))
+        except ValueError as e:
+            logger.error("Value error: %s", str(e))
+        return qs
+
+    def display_value(self, value):
+        """Convert stored numeric value to label for display."""
+        floor_plan_id = None
+        if not value:
+            return value
+
+        try:
+            if hasattr(self, "parent") and self.parent and hasattr(self.parent, "data"):
+                floor_plan_id = self.parent.data.get("nautobot_floor_plan_floor_plan")
+            if floor_plan_id and value is not None:
```

Comment:
Probably only need this since we already checked on line 56
```suggestion
            if floor_plan_id:
```